Add example /usr/bin/env python as Python interpreter

PierreBeucher opened the pull request:
I was looking for a way to use Python interpreter from an environment variable or directly from `PATH` rather than specifying absolute path as `ansible_python_interpreter`. For instance, I'm creating NixOS instance in Cloud and path to Python is hard to know in advance, I'd rather use the one from `PATH`.

Found out via https://github.com/ansible/ansible/issues/6345#issuecomment-181999529 one could use 

```
ansible_python_interpreter: '/usr/bin/env python'
```

And though it would be worth adding an example in doc :)

PierreBeucher committed:
Add example /usr/bin/env python as Python interpreter

Sometime users will want to use the first Python interpreter available on PATH and/or can't know in advance the Python interpreter absolute path on managed hosts (such as NixOS where path might change in Nix store).

Method found from this comment: https://github.com/ansible/ansible/issues/6345#issuecomment-181999529

gotmax23 commented:
Thanks for your Ansible docs contribution! We talk about Ansible documentation on matrix at [#docs:ansible.im](https://matrix.to/#/#docs:ansible.im) and on libera IRC at #ansible-docs if you ever want to join us and chat about the docs! We meet there on Tuesdays (see [the Ansible calendar](https://github.com/ansible/community/blob/main/meetings/README.md)) and welcome additions to our [weekly agenda items](https://github.com/ansible/community/issues/678) - scroll down to find the upcoming agenda and add a comment to put something new on that agenda.
<!--- boilerplate: docs_team_info --->

gotmax23 commented:
I''m not sure python_3_support.rst is the right place for this note. Interpreter discovery has its own doc. This point isn't specifically related to python3.

PierreBeucher commented:
I looked around a bit before adding an example here as it also contains other example usage of `ansible_python_interpreter`. Why would the sub-section _Using Python 3 on the managed machines with commands and playbooks_ not be suitable? It's related to using a specific Python interpreter for a managed machine. 

Another candidate would be https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html but it clearly references the Python 3 support page for `ansible_python_interpreter`, hence putting it here.

What would you think is a better page for this? I can update MR accordingly

oraNod commented on docs/docsite/rst/reference_appendices/python_3_support.rst:
```diff
@@ -68,6 +68,12 @@ Using Python 3 on the managed machines with commands and playbooks
     $ ansible localhost-py3 -m ping
     $ ansible-playbook sample-playbook.yml
 
+* If you want to use the first Python found on ``PATH`` or Python interpreter path can't be known in advance, you can use ``/usr/bin/env python`` such as:
```
```suggestion
* To use the first Python found on ``PATH`` or if the Python interpreter path is not known in advance, you can use ``/usr/bin/env python`` such as:
```

oraNod commented:
@PierreBeucher Thanks for the docs contribution. I added a comment with some minor wording suggestions. As far as where this should go in the docs, my 2c is that it might fit ok after the last sentence in `reference_appendices/interpreter_discovery.html`. It is a bit tricky to find the best spot though. :smile:

PierreBeucher commented:
> added a comment with some minor wording suggestions. 

Thanks, I updated with your proposed change.

How about a `seealso::` to `python_3_support.rst` from `interpreter_discovery.rst` for details and example about `ansible_python_interpreter` ? (it already exists the other way around so I guess it makes sense) This way all `ansible_python_interpreter` examples can be found at the same place, it will be less confusing for users and both "good" places are used. I changed accordingly

oraNod committed:
add sphinx anchor

oraNod reviewed (approved):
Pushed a commit to add a sphinx anchor for the Python 3 Support section. LGTM now. Thanks for the docs contribution @PierreBeucher

PierreBeucher commented:
Thanks @oraNod @samccann ! Glad this contribution made its way in :)
